wp-remember-the-galleries/wp-remember-the-galleries.php

```diff
@@ -308,50 +313,74 @@ static function save_gallery() {
 			wp_send_json_error( 'empty-name' );
 		}
 
+		$term_id = null; 
+		if ( isset( $_POST['term_id'] ) ) {
+			$term_id = (int) $_POST['term_id'];
+		}
+
 		$gallery_name = sanitize_text_field( trim( $_POST['name'] ) );
 
 		if( empty( $gallery_name ) ) {
 			wp_send_json_error( 'empty-name' );
 		}
 
-		// Get or insert the appropriate term for this gallery
+		// Does a gallery with this name already exist?
 		$term_info = term_exists( $gallery_name, self::entity );
+		$allow_rename = true;
+		if ( isset( $term_info['term_id'] ) && $term_info['term_id'] != $term_id ) {
+			// Ask user to confirm that they want to overwrite the existing gallery
+			if ( !isset( $_POST['yes'] ) ) {
+				wp_send_json_error( 'need-confirm' );
+			}
+			// User has confirmed they want to overwrite the existing gallery, so let's use that term id
+			else {
+				$allow_rename = false;
+				$term_id = $term_info['term_id'];
+			}
+		}
 
-		if( !isset( $term_info['term_id'] ) ) {
-			$term_info = wp_insert_term( $gallery_name, self::entity, array( 'slug' => self::entity . '--' ) );
+		if( !$term_id ) {
+			$term_info = wp_insert_term( $gallery_name, self::entity, array( 'slug' => self::entity . '-' . $post_id ) );
```

Comment:
Wow. Duh. Yeah I had that code **after** the post update/insert, but the rename still wasn't working right. I moved it above to fix the rename and broke the creation. (facepalm)